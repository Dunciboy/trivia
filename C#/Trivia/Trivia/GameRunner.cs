using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Trivia
{
    public class GameRunner
    {

        private static bool _gameHasNoWinner;

        public static void Main(String[] args)
        {
            Game aGame = new Game();

            aGame.AddPlayer("Chet");
            aGame.AddPlayer("Pat");
            aGame.AddPlayer("Sue");
            
            Random rand = (args.Length == 0 ? new Random() : new Random(args[0].GetHashCode()));

            do
            {

                aGame.MoveCurrentPlayer(rand.Next(5) + 1);

                if (rand.Next(9) == 7)
                {
                    _gameHasNoWinner = aGame.WrongAnswer();
                }
                else
                {
                    _gameHasNoWinner = aGame.WasCorrectlyAnswered();
                }



            } while (_gameHasNoWinner);

        }
    }

}

