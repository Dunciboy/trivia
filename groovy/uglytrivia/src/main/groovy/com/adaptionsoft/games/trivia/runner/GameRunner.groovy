package com.adaptionsoft.games.trivia.runner;

import com.adaptionsoft.games.uglytrivia.Game;

public class GameRunner {

	private static boolean notAWinner = true

	public static void main(String[] args) {
		def aGame = new Game()

		aGame.add("Chet")
		aGame.add("Pat")
		aGame.add("Sue")

		def rand = new Random()

		while (notAWinner) {
			aGame.roll(rand.nextInt(5) + 1)

			if (rand.nextInt(9) == 7) {
				notAWinner = aGame.wrongAnswer()
			} else {
				notAWinner = aGame.wasCorrectlyAnswered()
			}
		}
	}
}
