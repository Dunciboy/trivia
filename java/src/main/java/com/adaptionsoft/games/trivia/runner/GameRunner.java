
package com.adaptionsoft.games.trivia.runner;

import com.adaptionsoft.games.uglytrivia.Game;

import java.util.Random;


public class GameRunner {

    public static final int BOUND = 9;
    public static final int LOOSE = 7;
    public static final int MAXIMUM_EYES_ON_DICE = 5;

    private static Random rand = new Random();

    private static boolean notAWinner;

    public GameRunner(Random rand) {
        this.rand = rand;
    }

    public static void main(String[] args) {
        Game aGame = new Game();

        aGame.add("Chet");
        aGame.add("Pat");
        aGame.add("Sue");


        do {

            aGame.roll(rand.nextInt(MAXIMUM_EYES_ON_DICE) + 1);

            if (rand.nextInt(BOUND) == LOOSE) {
                notAWinner = aGame.wrongAnswer();
            } else {
                notAWinner = aGame.wasCorrectlyAnswered();
            }


        } while (notAWinner);

    }
}
