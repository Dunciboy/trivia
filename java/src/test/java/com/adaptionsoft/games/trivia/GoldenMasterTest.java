package com.adaptionsoft.games.trivia;

import com.adaptionsoft.games.trivia.runner.GameRunner;
import org.approvaltests.Approvals;
import org.junit.Test;

import java.io.ByteArrayOutputStream;
import java.io.PrintStream;
import java.util.Arrays;
import java.util.Random;

public class GoldenMasterTest {

    @Test
    public void verify_test_approval() throws Exception {
        String[] names = {"Llewellyn", "James", "Dan", "Jason", "Katrina"};
        Arrays.sort(names);
        Approvals.verifyAll("", names);
    }

    @Test
    public void verify_main() throws Exception {
        ByteArrayOutputStream baout = new ByteArrayOutputStream();
        PrintStream output = new PrintStream(baout);
        System.setOut(output);

        for (int index = 1; index < 5; index++) {
            Random random = new OurRandom(index);

            new GameRunner(random);
            GameRunner.main(new String[0]);
        }

        String[] actual = baout.toString("UTF-8").split("\\n");

        Approvals.verifyAll("", actual);
    }

    private class OurRandom extends Random {
        private int nextInt;

        public OurRandom(int nextInt) {
            this.nextInt = nextInt;
        }

        @Override
        public int nextInt(int bound) {
            return nextInt;
        }
    }

}




