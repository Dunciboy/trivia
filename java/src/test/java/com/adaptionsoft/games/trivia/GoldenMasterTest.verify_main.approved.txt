[000] = Chet was added
[001] = They are player number 1
[002] = Pat was added
[003] = They are player number 2
[004] = Sue was added
[005] = They are player number 3
[006] = Chet is the current player
[007] = They have rolled a 2
[008] = Chet's new location is 2
[009] = The category is Sports
[010] = Sports Question 0
[011] = Answer was correct!!!
[012] = Chet now has 1 Gold Coins.
[013] = Pat is the current player
[014] = They have rolled a 2
[015] = Pat's new location is 2
[016] = The category is Sports
[017] = Sports Question 1
[018] = Answer was correct!!!
[019] = Pat now has 1 Gold Coins.
[020] = Sue is the current player
[021] = They have rolled a 2
[022] = Sue's new location is 2
[023] = The category is Sports
[024] = Sports Question 2
[025] = Answer was correct!!!
[026] = Sue now has 1 Gold Coins.
[027] = Chet is the current player
[028] = They have rolled a 2
[029] = Chet's new location is 4
[030] = The category is Pop
[031] = Pop Question 0
[032] = Answer was correct!!!
[033] = Chet now has 2 Gold Coins.
[034] = Pat is the current player
[035] = They have rolled a 2
[036] = Pat's new location is 4
[037] = The category is Pop
[038] = Pop Question 1
[039] = Answer was correct!!!
[040] = Pat now has 2 Gold Coins.
[041] = Sue is the current player
[042] = They have rolled a 2
[043] = Sue's new location is 4
[044] = The category is Pop
[045] = Pop Question 2
[046] = Answer was correct!!!
[047] = Sue now has 2 Gold Coins.
[048] = Chet is the current player
[049] = They have rolled a 2
[050] = Chet's new location is 6
[051] = The category is Sports
[052] = Sports Question 3
[053] = Answer was correct!!!
[054] = Chet now has 3 Gold Coins.
[055] = Pat is the current player
[056] = They have rolled a 2
[057] = Pat's new location is 6
[058] = The category is Sports
[059] = Sports Question 4
[060] = Answer was correct!!!
[061] = Pat now has 3 Gold Coins.
[062] = Sue is the current player
[063] = They have rolled a 2
[064] = Sue's new location is 6
[065] = The category is Sports
[066] = Sports Question 5
[067] = Answer was correct!!!
[068] = Sue now has 3 Gold Coins.
[069] = Chet is the current player
[070] = They have rolled a 2
[071] = Chet's new location is 8
[072] = The category is Pop
[073] = Pop Question 3
[074] = Answer was correct!!!
[075] = Chet now has 4 Gold Coins.
[076] = Pat is the current player
[077] = They have rolled a 2
[078] = Pat's new location is 8
[079] = The category is Pop
[080] = Pop Question 4
[081] = Answer was correct!!!
[082] = Pat now has 4 Gold Coins.
[083] = Sue is the current player
[084] = They have rolled a 2
[085] = Sue's new location is 8
[086] = The category is Pop
[087] = Pop Question 5
[088] = Answer was correct!!!
[089] = Sue now has 4 Gold Coins.
[090] = Chet is the current player
[091] = They have rolled a 2
[092] = Chet's new location is 10
[093] = The category is Sports
[094] = Sports Question 6
[095] = Answer was correct!!!
[096] = Chet now has 5 Gold Coins.
[097] = Pat is the current player
[098] = They have rolled a 2
[099] = Pat's new location is 10
[100] = The category is Sports
[101] = Sports Question 7
[102] = Answer was correct!!!
[103] = Pat now has 5 Gold Coins.
[104] = Sue is the current player
[105] = They have rolled a 2
[106] = Sue's new location is 10
[107] = The category is Sports
[108] = Sports Question 8
[109] = Answer was correct!!!
[110] = Sue now has 5 Gold Coins.
[111] = Chet is the current player
[112] = They have rolled a 2
[113] = Chet's new location is 0
[114] = The category is Pop
[115] = Pop Question 6
[116] = Answer was correct!!!
[117] = Chet now has 6 Gold Coins.
[118] = Chet was added
[119] = They are player number 1
[120] = Pat was added
[121] = They are player number 2
[122] = Sue was added
[123] = They are player number 3
[124] = Chet is the current player
[125] = They have rolled a 3
[126] = Chet's new location is 3
[127] = The category is Rock
[128] = Rock Question 0
[129] = Answer was correct!!!
[130] = Chet now has 1 Gold Coins.
[131] = Pat is the current player
[132] = They have rolled a 3
[133] = Pat's new location is 3
[134] = The category is Rock
[135] = Rock Question 1
[136] = Answer was correct!!!
[137] = Pat now has 1 Gold Coins.
[138] = Sue is the current player
[139] = They have rolled a 3
[140] = Sue's new location is 3
[141] = The category is Rock
[142] = Rock Question 2
[143] = Answer was correct!!!
[144] = Sue now has 1 Gold Coins.
[145] = Chet is the current player
[146] = They have rolled a 3
[147] = Chet's new location is 6
[148] = The category is Sports
[149] = Sports Question 0
[150] = Answer was correct!!!
[151] = Chet now has 2 Gold Coins.
[152] = Pat is the current player
[153] = They have rolled a 3
[154] = Pat's new location is 6
[155] = The category is Sports
[156] = Sports Question 1
[157] = Answer was correct!!!
[158] = Pat now has 2 Gold Coins.
[159] = Sue is the current player
[160] = They have rolled a 3
[161] = Sue's new location is 6
[162] = The category is Sports
[163] = Sports Question 2
[164] = Answer was correct!!!
[165] = Sue now has 2 Gold Coins.
[166] = Chet is the current player
[167] = They have rolled a 3
[168] = Chet's new location is 9
[169] = The category is Science
[170] = Science Question 0
[171] = Answer was correct!!!
[172] = Chet now has 3 Gold Coins.
[173] = Pat is the current player
[174] = They have rolled a 3
[175] = Pat's new location is 9
[176] = The category is Science
[177] = Science Question 1
[178] = Answer was correct!!!
[179] = Pat now has 3 Gold Coins.
[180] = Sue is the current player
[181] = They have rolled a 3
[182] = Sue's new location is 9
[183] = The category is Science
[184] = Science Question 2
[185] = Answer was correct!!!
[186] = Sue now has 3 Gold Coins.
[187] = Chet is the current player
[188] = They have rolled a 3
[189] = Chet's new location is 0
[190] = The category is Pop
[191] = Pop Question 0
[192] = Answer was correct!!!
[193] = Chet now has 4 Gold Coins.
[194] = Pat is the current player
[195] = They have rolled a 3
[196] = Pat's new location is 0
[197] = The category is Pop
[198] = Pop Question 1
[199] = Answer was correct!!!
[200] = Pat now has 4 Gold Coins.
[201] = Sue is the current player
[202] = They have rolled a 3
[203] = Sue's new location is 0
[204] = The category is Pop
[205] = Pop Question 2
[206] = Answer was correct!!!
[207] = Sue now has 4 Gold Coins.
[208] = Chet is the current player
[209] = They have rolled a 3
[210] = Chet's new location is 3
[211] = The category is Rock
[212] = Rock Question 3
[213] = Answer was correct!!!
[214] = Chet now has 5 Gold Coins.
[215] = Pat is the current player
[216] = They have rolled a 3
[217] = Pat's new location is 3
[218] = The category is Rock
[219] = Rock Question 4
[220] = Answer was correct!!!
[221] = Pat now has 5 Gold Coins.
[222] = Sue is the current player
[223] = They have rolled a 3
[224] = Sue's new location is 3
[225] = The category is Rock
[226] = Rock Question 5
[227] = Answer was correct!!!
[228] = Sue now has 5 Gold Coins.
[229] = Chet is the current player
[230] = They have rolled a 3
[231] = Chet's new location is 6
[232] = The category is Sports
[233] = Sports Question 3
[234] = Answer was correct!!!
[235] = Chet now has 6 Gold Coins.
[236] = Chet was added
[237] = They are player number 1
[238] = Pat was added
[239] = They are player number 2
[240] = Sue was added
[241] = They are player number 3
[242] = Chet is the current player
[243] = They have rolled a 4
[244] = Chet's new location is 4
[245] = The category is Pop
[246] = Pop Question 0
[247] = Answer was correct!!!
[248] = Chet now has 1 Gold Coins.
[249] = Pat is the current player
[250] = They have rolled a 4
[251] = Pat's new location is 4
[252] = The category is Pop
[253] = Pop Question 1
[254] = Answer was correct!!!
[255] = Pat now has 1 Gold Coins.
[256] = Sue is the current player
[257] = They have rolled a 4
[258] = Sue's new location is 4
[259] = The category is Pop
[260] = Pop Question 2
[261] = Answer was correct!!!
[262] = Sue now has 1 Gold Coins.
[263] = Chet is the current player
[264] = They have rolled a 4
[265] = Chet's new location is 8
[266] = The category is Pop
[267] = Pop Question 3
[268] = Answer was correct!!!
[269] = Chet now has 2 Gold Coins.
[270] = Pat is the current player
[271] = They have rolled a 4
[272] = Pat's new location is 8
[273] = The category is Pop
[274] = Pop Question 4
[275] = Answer was correct!!!
[276] = Pat now has 2 Gold Coins.
[277] = Sue is the current player
[278] = They have rolled a 4
[279] = Sue's new location is 8
[280] = The category is Pop
[281] = Pop Question 5
[282] = Answer was correct!!!
[283] = Sue now has 2 Gold Coins.
[284] = Chet is the current player
[285] = They have rolled a 4
[286] = Chet's new location is 0
[287] = The category is Pop
[288] = Pop Question 6
[289] = Answer was correct!!!
[290] = Chet now has 3 Gold Coins.
[291] = Pat is the current player
[292] = They have rolled a 4
[293] = Pat's new location is 0
[294] = The category is Pop
[295] = Pop Question 7
[296] = Answer was correct!!!
[297] = Pat now has 3 Gold Coins.
[298] = Sue is the current player
[299] = They have rolled a 4
[300] = Sue's new location is 0
[301] = The category is Pop
[302] = Pop Question 8
[303] = Answer was correct!!!
[304] = Sue now has 3 Gold Coins.
[305] = Chet is the current player
[306] = They have rolled a 4
[307] = Chet's new location is 4
[308] = The category is Pop
[309] = Pop Question 9
[310] = Answer was correct!!!
[311] = Chet now has 4 Gold Coins.
[312] = Pat is the current player
[313] = They have rolled a 4
[314] = Pat's new location is 4
[315] = The category is Pop
[316] = Pop Question 10
[317] = Answer was correct!!!
[318] = Pat now has 4 Gold Coins.
[319] = Sue is the current player
[320] = They have rolled a 4
[321] = Sue's new location is 4
[322] = The category is Pop
[323] = Pop Question 11
[324] = Answer was correct!!!
[325] = Sue now has 4 Gold Coins.
[326] = Chet is the current player
[327] = They have rolled a 4
[328] = Chet's new location is 8
[329] = The category is Pop
[330] = Pop Question 12
[331] = Answer was correct!!!
[332] = Chet now has 5 Gold Coins.
[333] = Pat is the current player
[334] = They have rolled a 4
[335] = Pat's new location is 8
[336] = The category is Pop
[337] = Pop Question 13
[338] = Answer was correct!!!
[339] = Pat now has 5 Gold Coins.
[340] = Sue is the current player
[341] = They have rolled a 4
[342] = Sue's new location is 8
[343] = The category is Pop
[344] = Pop Question 14
[345] = Answer was correct!!!
[346] = Sue now has 5 Gold Coins.
[347] = Chet is the current player
[348] = They have rolled a 4
[349] = Chet's new location is 0
[350] = The category is Pop
[351] = Pop Question 15
[352] = Answer was correct!!!
[353] = Chet now has 6 Gold Coins.
[354] = Chet was added
[355] = They are player number 1
[356] = Pat was added
[357] = They are player number 2
[358] = Sue was added
[359] = They are player number 3
[360] = Chet is the current player
[361] = They have rolled a 5
[362] = Chet's new location is 5
[363] = The category is Science
[364] = Science Question 0
[365] = Answer was correct!!!
[366] = Chet now has 1 Gold Coins.
[367] = Pat is the current player
[368] = They have rolled a 5
[369] = Pat's new location is 5
[370] = The category is Science
[371] = Science Question 1
[372] = Answer was correct!!!
[373] = Pat now has 1 Gold Coins.
[374] = Sue is the current player
[375] = They have rolled a 5
[376] = Sue's new location is 5
[377] = The category is Science
[378] = Science Question 2
[379] = Answer was correct!!!
[380] = Sue now has 1 Gold Coins.
[381] = Chet is the current player
[382] = They have rolled a 5
[383] = Chet's new location is 10
[384] = The category is Sports
[385] = Sports Question 0
[386] = Answer was correct!!!
[387] = Chet now has 2 Gold Coins.
[388] = Pat is the current player
[389] = They have rolled a 5
[390] = Pat's new location is 10
[391] = The category is Sports
[392] = Sports Question 1
[393] = Answer was correct!!!
[394] = Pat now has 2 Gold Coins.
[395] = Sue is the current player
[396] = They have rolled a 5
[397] = Sue's new location is 10
[398] = The category is Sports
[399] = Sports Question 2
[400] = Answer was correct!!!
[401] = Sue now has 2 Gold Coins.
[402] = Chet is the current player
[403] = They have rolled a 5
[404] = Chet's new location is 3
[405] = The category is Rock
[406] = Rock Question 0
[407] = Answer was correct!!!
[408] = Chet now has 3 Gold Coins.
[409] = Pat is the current player
[410] = They have rolled a 5
[411] = Pat's new location is 3
[412] = The category is Rock
[413] = Rock Question 1
[414] = Answer was correct!!!
[415] = Pat now has 3 Gold Coins.
[416] = Sue is the current player
[417] = They have rolled a 5
[418] = Sue's new location is 3
[419] = The category is Rock
[420] = Rock Question 2
[421] = Answer was correct!!!
[422] = Sue now has 3 Gold Coins.
[423] = Chet is the current player
[424] = They have rolled a 5
[425] = Chet's new location is 8
[426] = The category is Pop
[427] = Pop Question 0
[428] = Answer was correct!!!
[429] = Chet now has 4 Gold Coins.
[430] = Pat is the current player
[431] = They have rolled a 5
[432] = Pat's new location is 8
[433] = The category is Pop
[434] = Pop Question 1
[435] = Answer was correct!!!
[436] = Pat now has 4 Gold Coins.
[437] = Sue is the current player
[438] = They have rolled a 5
[439] = Sue's new location is 8
[440] = The category is Pop
[441] = Pop Question 2
[442] = Answer was correct!!!
[443] = Sue now has 4 Gold Coins.
[444] = Chet is the current player
[445] = They have rolled a 5
[446] = Chet's new location is 1
[447] = The category is Science
[448] = Science Question 3
[449] = Answer was correct!!!
[450] = Chet now has 5 Gold Coins.
[451] = Pat is the current player
[452] = They have rolled a 5
[453] = Pat's new location is 1
[454] = The category is Science
[455] = Science Question 4
[456] = Answer was correct!!!
[457] = Pat now has 5 Gold Coins.
[458] = Sue is the current player
[459] = They have rolled a 5
[460] = Sue's new location is 1
[461] = The category is Science
[462] = Science Question 5
[463] = Answer was correct!!!
[464] = Sue now has 5 Gold Coins.
[465] = Chet is the current player
[466] = They have rolled a 5
[467] = Chet's new location is 6
[468] = The category is Sports
[469] = Sports Question 3
[470] = Answer was correct!!!
[471] = Chet now has 6 Gold Coins.
